### ISSUE TITLE
stack encoder layer for transformer model

### ISSUE THREAD

**Issue by Yingzhu-Zhao** [espnet/espnet#1278]
Hi,

I see for multiple encoder layers in transformer encoder model, you use repeat.py function to repeat encoder_layer.py. However, I modify the code to be below, the result is very bad. Do you know why?

self.encoders = EncoderLayer(
            attention_dim,
            MultiHeadedAttention(attention_heads, attention_dim, attention_dropout_rate),
            PositionwiseFeedForward(attention_dim, linear_units, dropout_rate),
            dropout_rate,
            normalize_before,
            concat_after
        )

for i in range(12):
            xs, masks = self.encoders(xs, masks)

Basically I remove the repeat function, and in the forward function, use loop 12 to stack 12 encoder layers. There is no other change. The result is very bad.

Thanks.

**Comment by kan-bayashi** [comment]
It seems that the same layer is used 12 times.

**Comment by Yingzhu-Zhao** [comment]
Hi, thanks for your reply. But for repeat.py function, isn't it repeat the same layer 12 times?

**Comment by kan-bayashi** [comment]
No. We define the layer 12 times.
Please carefully check the code.
https://github.com/espnet/espnet/blob/master/espnet/nets/pytorch_backend/transformer/repeat.py

**Comment by mowayao** [comment]
Cross-layer parameter sharing is not working on ASR?
